libyul/AsmParser.cpp

```diff
@@ -224,30 +259,45 @@ std::optional<std::pair<std::string_view, SourceLocation>> Parser::parseSrcComme
 		return {{tail, SourceLocation{}}};
 	}
 
-	std::optional<int> const sourceIndex = toInt(match[1].str());
-	std::optional<int> const start = toInt(match[2].str());
-	std::optional<int> const end = toInt(match[3].str());
+	int sourceIndex = -1;
+	int start = -1;
+	int end = -1;
+	bool const successfullyInterpretedIndices = [&]()
+	{
+		if (successfullyScannedIndices)
+		{
+			auto const indexPairs = { std::tie(rawSourceIndex, sourceIndex), std::tie(rawStart, start), std::tie(rawEnd, end) };
+			for (auto& [rawIndex, interpretedIndex] : indexPairs)
+			{
+				auto const interpreted = toInt(*rawIndex);
+				if (interpreted)
+					interpretedIndex = *interpreted;
```

Comment:
This kind of naming is one of my pet peeves :P Good names for similar (but not identical) things should really focus on what distinguishes them so that you can tell it without too much thinking. Something like this would be much clearer IMO:

```suggestion
				auto const indexOrNull = toInt(*rawIndex);
				if (indexOrNull)
					index = *indexOrNull;
```
This is similar to the `val` vs `value` situation.

Reply:
Also, I think that the whole conversion code here would really be better off if it was much less clever :)

```c++
std::optional<int> const sourceIndex = toInt(*rawSourceIndex);
std::optional<int> const start = toInt(*startIndex);
std::optional<int> const end = toInt(*endIndex);

if (
	!sourceIndex.has_value() || *sourceIndex < -1 ||
	!start.has_value() || *start < -1 ||
	!end.has_value() || *end < -1
)
```

This is more repetitive but also IMO much easier to understand quickly. Especially modifying values through local references makes code more confusing for me. There's too much indirection.

By the way, note that you don't have to check `successfullyScannedIndices` because you already made sure above that it is true.

Reply:
Yeah you are right, this reads much better. Guess I was trying to be too clever at the expense of readability :)